### PR TITLE
network: clean-up DHCP lease server data configuration

### PR DESCRIPTION
**Pull request by poettering**
This is an attempt to clean up the POP3/SMTP/LPR/… DHCP lease server
data logic in networkd. This reduces code duplication and fixes a number
of bugs.

This removes any support for collecting POP3/SMPT/LPR servers acquired
via local DHCP client releases since noone uses that, and given how old
these protocols are I doubt this will change. It keeps support for
configuring them for the dhcp server however.

The differences between the DNS/NTP/SIP/POP3/SMTP/LPR configuration
logics are minimized.

This removes the relevant symbols from sd-network.h (which is an
internal API only at this point after all).

This is unfortunately not well test, given the old code for this had
barely any tests. But the new code should not perform worse at least,
and allow us to release, since it corrects some interfaces visible in
the .network configuration format.

Fixes: #15943

### CODE REVIEW

File: NEWS [hunk 1]
```diff
@@ -76,8 +76,8 @@ CHANGES WITH 246 in spe:
           field.
 
         * systemd-networkd's [DHCPServer] section gained a new set of options
-          POP3Servers=, SMTPServers=, LPRServers= for including server
-          information about these three protocols in the DHCP lease. It also
+          EmitPOP3=/POP3=, EmitSMTP=/SMTP=, EmitLPR=/LPR= for including server
+          information about these three protocols in theS DHCP lease. It also
```
Comment (dnicolodi) [inline NEWS]: typo: `s/theS/the/`

Comment (yuwata) [reply NEWS]: Fixed.

**Comment by yuwata** [comment]
I've force pushed an updated version, which addresses the above typo, and fixes `reallocarray()` issue found by the fuzzers.

**Comment by ssahani** [comment]
Everything is green and fixed . I think this can be merged.

**Pull request merged**